core/src/main/java/com/linecorp/armeria/server/annotation/JavaBeanRequestConverterFunction.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation;
+
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.convertParameter;
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.httpParametersOf;
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.validateAndNormalizeSupportedType;
+import static org.reflections.ReflectionUtils.getAllFields;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpParameters;
+import com.linecorp.armeria.common.util.Holder;
+import com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.EnumConverter;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.util.AsciiString;
+
+/**
+ * A default implementation of a {@link RequestConverterFunction} which converts the following parameters
+ * of the {@link AggregatedHttpMessage} to an object.
+ *
+ * <p>Path parameters to fields annotated by {@link Param}
+ *
+ * <p>HTTP parameters to fields annotated by {@link Param}
+ *
+ * <p>Headers to fields annotated by {@link Header}
+ */
+public class JavaBeanRequestConverterFunction implements RequestConverterFunction {
+    private static final ConcurrentMap<Class<?>, BeanParamInfo> beanParamInfoCache = new ConcurrentHashMap<>();
+
+    private static final List<Class<?>> ignoredResultTypeList = new ArrayList<>();
```

Comment:
Both `beanParamInfoCache` and `ignoredResultTypeList` must have weak or phantom reference key, because otherwise they can lead to class loader leak in a container environment. You could use Guava's `MapMaker` for `beanParamInfoCache` for instance.

Reply:
fixed.

But looks like there is not something like `WeakReferenceList`, so I renamed `ignoredResultTypeList ` to `ignoredResultTypeCache`, and use `new MapMaker().weakKeys().makeMap()` for it too